### PR TITLE
refactor: changed how DI is initialized

### PR DESCRIPTION
**Pull request by CptSchnitz**
reworked how DI is initialized, so there is no need to redo it for tests.

### CODE REVIEW

File: src/resourceName/routes/resourceNameRouter.ts [hunk 1]
```diff
@@ -12,4 +12,6 @@ const resourceNameRouterFactory: FactoryFunction<Router> = (dependencyContainer)
   return router;
 };
 
+export const resourceNameRouterSymbol = Symbol('resourceNameRouterFactory');
```
Comment (syncush) [inline src/resourceName/routes/resourceNameRouter.ts:15]: I think it should be capped as it is a const

**Review by syncush — CHANGES_REQUESTED** [review]
maybe we should add one more resource, to demonstrate how it should work when there are more than 1 reosurce

File: tests/integration/resourceName/resourceName.spec.ts [hunk 2]
```diff
@@ -1,17 +1,25 @@
...
+        { token: Services.LOGGER, provider: { useValue: jsLogger({ enabled: false }) } },
+        { token: Services.TRACER, provider: { useValue: trace.getTracer('testTracer') } },
+      ],
+      useChild: true,
+    });
+    requestSender = new ResourceNameRequestSender(app);
   });
   afterEach(function () {
-    container.clearInstances();
+    // container.clearInstances();
```
Comment (asafMasa) [inline tests/integration/resourceName/resourceName.spec.ts:22]: remove remark

**Review by asafMasa — APPROVED** [review]

File: tests/integration/resourceName/resourceName.spec.ts (hunk 2 above)
Comment (CptSchnitz) [reply tests/integration/resourceName/resourceName.spec.ts]: removed

File: src/resourceName/routes/resourceNameRouter.ts (hunk 1 above)
Comment (CptSchnitz) [reply src/resourceName/routes/resourceNameRouter.ts]: capped

File: src/anotherResource/controllers/anotherResourceController.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import { Meter } from '@map-colonies/telemetry';
+import { BoundCounter } from '@opentelemetry/api-metrics';
+import { RequestHandler } from 'express';
+import httpStatus from 'http-status-codes';
+import { injectable, inject } from 'tsyringe';
+import { Services } from '../../common/constants';
+
+import { IAnotherResourceModel, AnotherResourceManager } from '../models/anotherResourceManager';
+
+type CreateResourceHandler = RequestHandler<undefined, IAnotherResourceModel, IAnotherResourceModel>;
```
Comment (syncush) [inline src/anotherResource/controllers/anotherResourceController.ts:11]: defined but never used

**Review by syncush — CHANGES_REQUESTED** [review]

File: src/anotherResource/controllers/anotherResourceController.ts (hunk 3 above)
Comment (CptSchnitz) [reply src/anotherResource/controllers/anotherResourceController.ts]: fixed

**Review by syncush — APPROVED** [review]

**Review by syncush — APPROVED** [review]

**Pull request merged**